### PR TITLE
additional logic added to export only certain columns

### PR DESCRIPTION
**Pull request by AEscobarCruz**
Should help in performing region coding much more efficiently and with less memory constraints than loading full rows to memory.

### CODE REVIEW

File: soda-fountain-lib/src/main/scala/com/socrata/soda/server/resources/Export.scala [hunk 1]
```diff
@@ -82,6 +82,34 @@ case class Export(exportDAO: ExportDAO, etagObfuscator: ETagObfuscator) {
...
+    // get only these columns, we expect these to be fieldnames comma separated items.
+    val columnsOnly = Option(req.getParameter("columns")).map {
+      paramStr =>
+        try {
+          val columnFields = paramStr.toLowerCase.split(",").toSeq.map(x => x.trim)
+          val columns = exportDAO.lookupColumns(resourceName, Stage(copy)).getOrElse(Seq.empty)
+          // need to have the row-identifier included otherwise dc will fail
+          // this may seem long winded but helps avoid double counting in case one of requested
+          // columns is the row-identifier
+          val rowIdColumn = columns.filter{c => c.typ == SoQLID}.map{c => c.fieldName.name}
```
Comment (NathalieKaligirwa) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/resources/Export.scala]: Is SoQLID, system column `:id`?

Comment (AEscobarCruz) [inline soda-fountain-lib/src/main/scala/com/socrata/soda/server/resources/Export.scala]: can be, i believe and any columned labeled as a row-identifier.

**Comment by AEscobarCruz** [comment]
The store object is not available at the export level only in the doa level should it even be exposed, no?

**Comment by chitang** [comment]
That's right.  :shipit:

**Pull request merged**